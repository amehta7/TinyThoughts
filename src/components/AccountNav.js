import React, { memo } from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'

const AccountNav = memo(({ user }) => {
  return (
    <div className='account-nav'>
      {' '}
      {!user ? (
        <ul>
          <li>
            <Link to='/signin'>Signin</Link>
          </li>
          <li>
            {' '}
            <Link to='/signup'>Signup</Link>
          </li>
        </ul>
      ) : (
        <React.Fragment>
          <h2>{user.name}</h2>
          <ul>
            <li>
              <Link to='/'>Signout</Link>
            </li>
          </ul>
        </React.Fragment>
      )}
    </div>
  )
})

const mapStateToProps = (state) => {
  return {
    user: state.users.user,
  }
}

export default connect(mapStateToProps, null)(AccountNav)

// <h4>user_name</h4>
// <ul>
//   <li>Signout</li>
// </ul>
