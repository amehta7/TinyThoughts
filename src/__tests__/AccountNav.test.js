import React from "react";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import toJson from "enzyme-to-json";
import thunk from "redux-thunk";
import AccountNav from "../components/AccountNav";
const mockStore = configureStore([thunk]);

describe("AccountNav component", () => {
  it("Renders the Signin and Signup buttons when logged out", () => {
    const store = mockStore({
      users: {}
    });
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <AccountNav />
        </MemoryRouter>
      </Provider>
    );

    return expect(wrapper.find(AccountNav)).toMatchSnapshot();
  });

  it("Renders the Signout button & user's name when logged in", () => {
    const store = mockStore({
      users: {
        user: { id: 1, name: "Joe Mockery" }
      }
    });
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <AccountNav />
        </MemoryRouter>
      </Provider>
    );

    return expect(wrapper.find(AccountNav)).toMatchSnapshot();
  });
});
