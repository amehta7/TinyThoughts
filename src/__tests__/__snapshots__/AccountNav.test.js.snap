// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountNav component Renders the Signin and Signup buttons when logged out 1`] = `
<Memo()>
  <div
    className="account-nav"
  >
    <ul>
      <Link
        to="/signin"
      >
        <LinkAnchor
          href="/signin"
          navigate={[Function]}
        >
          <a
            href="/signin"
            onClick={[Function]}
          >
            <li>
              Signin
            </li>
          </a>
        </LinkAnchor>
      </Link>
      <Link
        to="/signup"
      >
        <LinkAnchor
          href="/signup"
          navigate={[Function]}
        >
          <a
            href="/signup"
            onClick={[Function]}
          >
            <li>
              Signup
            </li>
          </a>
        </LinkAnchor>
      </Link>
    </ul>
  </div>
</Memo()>
`;

exports[`AccountNav component Renders the Signout button & user's name when logged in 1`] = `
<Memo()>
  <div
    className="account-nav"
  >
    <h4>
      Joe Mockery
    </h4>
    <ul>
      <li
        onClick={[Function]}
      >
        Signout
      </li>
    </ul>
  </div>
</Memo()>
`;
